### PR TITLE
docs: generate website documents correctly for 2.0

### LINKED ISSUE #1445: frontmatter example is not working
### Subject of the issue

See example https://mdxjs.com/advanced/api#compiler, it's expected to yield `{ title: "Hello, MDX" }` while I got `undefined`.

### Your environment

*   **OS**: macOS BigSur
*   **Packages**: 
*   **Env**: 

### Steps to reproduce

1. clone `https://github.com/chenxsan/bug-mdxjs-frontmatter`
2. run `yarn`
3. run `node index.js`

### Expected behaviour

Yields `{ title: "Hello, MDX" }`.

### Actual behaviour

Got `undefined`.

### PR DESCRIPTION
**Pull request by JounQin**
1. document remark-frontmatter@3.0 correctly for 2.0, close #1445
2. enable `eslint-plugin-mdx` correctly for `md` and `mdx` files
3. fix incorrect mdx version in website header
4. use `patch-package` for patching

cc @ChristianMurphy

### CODE REVIEW

**Comment by JounQin** [comment]
@ChristianMurphy Any luck to continue reviewing `patch-package`?

File: packages/gatsby-theme-mdx/src/pages/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "overrides": [
+    {
+      "files": "./conf.mdx",
+      "globals": {
+        "Speaker": false,
+        "SpeakerList": false
```
Comment (ChristianMurphy) [inline packages/gatsby-theme-mdx/src/pages/.eslintrc:7]:
```suggestion
        "Speaker": "readonly",
        "SpeakerList": "readonly"
```

https://eslint.org/docs/user-guide/configuring/language-options#using-configuration-files-1

**Review by ChristianMurphy — COMMENTED** [review]
Thanks @JounQin!

File: packages/gatsby-theme-mdx/src/pages/.eslintrc (hunk 1 above)
Comment (JounQin) [reply packages/gatsby-theme-mdx/src/pages/.eslintrc:7]: `false` means `readonly` too actually

Comment (JounQin) [reply packages/gatsby-theme-mdx/src/pages/.eslintrc]: @ChristianMurphy Done!

File: .eslintrc.yml [hunk 2]
```diff
@@ -43,13 +47,13 @@ settings:
...
 overrides:
-  - files: '*.mdx'
+  - files: '*.{md,mdx}'
     extends:
       - plugin:mdx/recommended
       - plugin:mdx/overrides
     rules:
       import/no-extraneous-dependencies: 0
-      prettier/prettier: 0
+      prettier/prettier: 2
```
Comment (ChristianMurphy) [inline .eslintrc.yml:56]: it would be good to use `off` and `error` for consistency with the rest of the config and clarity

Comment (JounQin) [inline .eslintrc.yml:56]:
```suggestion
      prettier/prettier: error
```

The original value is `0`, so I changed it to `2` without a second thought. 😂

Comment (JounQin) [reply .eslintrc.yml]: Done!

**Comment by JounQin** [comment]
@ChristianMurphy Can you approve and merge then?

**Comment by ChristianMurphy** [comment]
Thanks @JounQin, I appreciate you taking the time to carry this through. :bow: 
No objections from me, looks good. :+1: 

Though it would be nice to have another member of @mdx-js/mergers take a look as well, if any have time in the near future.

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
P.S., @JounQin, please split your PRs up more when possible in the future, so that it’s easier to review certain aspects and what they do

**Review by ChristianMurphy — APPROVED** [review]

**Pull request merged**